tests/common.rs

```diff
@@ -0,0 +1,86 @@
+// person obtaining a copy of this software, associated documentation and/or
+// data (collectively the "Software"), free of charge and under any and all
+// copyright rights in the Software, and any and all patent rights owned or
+// freely licensable by each licensor hereunder covering either (i) the
+// unmodified Software as contributed to or provided by such licensor, or (ii)
+// the Larger Works (as defined below), to deal in both
+//
+// (a) the Software, and
+// (b) any piece of software and/or hardware listed in the lrgrwrks.txt file
+// if one is included with the Software (each a "Larger Work" to which the Software
+// is contributed by such licensors),
+//
+// without restriction, including without limitation the rights to copy, create
+// derivative works of, display, perform, and distribute the Software and make,
+// use, sell, offer for sale, import, export, have made, and have sold the
+// Software and the Larger Work(s), and to sublicense the foregoing rights on
+// either these or other terms.
+//
+// This license is subject to the following condition: The above copyright
+// notice and either this complete permission notice or at a minimum a reference
+// to the UPL must be included in all copies or substantial portions of the
+// Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+//! Code common to several integration tests.
+#![warn(missing_docs)]
+
+extern crate diffract;
+
+use diffract::edit_script::{Chawathe96Config, EditScriptGenerator};
+use diffract::matchers::MatchTrees;
+use diffract::parser::{get_lexer, get_parser, parse_file};
+
+/// Check that the Chawathe 1996 edit script generator correctly generates an
+/// edit script for two files and a given matcher.
+pub fn check_files(path1: &str, path2: &str, matcher: Box<MatchTrees<String>>) {
+    // Lex and parse the input files.
+    let ast_from = parse_file(path1, &get_lexer(path1), &get_parser(path1)).unwrap();
+    let ast_to = parse_file(path2, &get_lexer(path2), &get_parser(path2)).unwrap();
+    // Generate mappings between ASTs.
+    let store = matcher.match_trees(ast_from.clone(), ast_to.clone());
+    // Generate an edit script.
+    let gen_config: Box<EditScriptGenerator<String>> = Box::new(Chawathe96Config::new());
+    let edit_script_wrapped = gen_config.generate_script(&store);
+    assert!(edit_script_wrapped.is_ok(),
+            "Edit script generator failed to complete.");
+    // Get root nodes after the edit script has run (note that the root nodes
```

Comment:
Are these the root nodes or root node IDs? I assume the latter, otherwise I'm not sure how anything could be mutated.

Reply:
They are root node ids. To avoid all sorts of fun with the borrow checker, many methods in `ast.rs` are methods on IDs, which pass the arena (the AST) to the method.